specs/waku/v2/waku-v2-rpc-api.md

```diff
@@ -0,0 +1,459 @@
+---
+title: Waku v2 JSON-RPC REST API
+version: 0.0.1
+status: Raw
+authors: Hanno Cornelius <hanno@status.im>
+---
+
+# Table of Contents
+
+1. [Introduction](#introduction)
+1. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport)
+    1. [Types](#types)
+    1. [Method naming](#method-naming)
+    1. [Debug API](#debug-api)
+    1. [Relay API](#relay-api)
+    1. [Store API](#store-api)
+    1. [Filter API](#filter-api)
+1. [Example usage](#example-usage)
+1. [Changelog](#changelog)
+1. [Copyright](#copyright)
+
+# Introduction
+
+This specification describes the JSON-RPC API that Waku v2 nodes MAY adhere to. Refer to the [Waku v2 specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-v2.md) for more information on Waku v2.
+
+# Wire Protocol
+
+## Transport
+
+Nodes SHOULD expose an accessible [JSON-RPC](https://www.jsonrpc.org/specification) API. The JSON-RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"get_waku_v2_debug_info",
+  "params":[],
+  "id":1
+}
+```
+
+### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON-RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON-RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
```

Comment:
What is the usage of `id`?

Reply:
So `id` is a "request" identifier, generated by the RPC client, that is included in the server response to tie the request and response together. Its usage is as per [JSON-RPC specification](https://www.jsonrpc.org/specification). I'm going to follow Oskar's suggestion below to include an explicit reference list, which will hopefully make it a bit clearer. :)